### PR TITLE
Fix ci build master

### PR DESCRIPTION
**Pull request by bikallem**
Address some more 32 warnings.

### CODE REVIEW

**Comment by bikallem** [comment]
@mseri I tracked the issue with `dune build` giving error/warning 32 in ci builds. It seems dune 3.0/3.1 raises this error while dune < 2.9 seems to not surface this error. eio-3 branch uses dune version 3.0 which is why the ci was failing in the branch but passing in the master. This PR addresses this issue with dune version and removes all instances of warning 32 in the code base.

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -45,6 +45,8 @@ jobs:
 
       - run: opam exec -- make build
 
+      - run: opam exec -- dune build bench/ examples/ cohttp-curl-async/bin/ cohttp-curl-lwt/bin/
```
Comment (rgrinberg) [inline .github/workflows/workflow.yml:48]: What additional targets does this add over `make build`?

File: CHANGES.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ## Unreleased
+- Address warning 32 errors in ci (@bikallem #876)
```
Comment (rgrinberg) [inline CHANGES.md:2]: We don't to describe CI changes in the change log.

File: .github/workflows/workflow.yml (hunk 1 above)
Comment (bikallem) [reply .github/workflows/workflow.yml]: removed.

File: CHANGES.md (hunk 2 above)
Comment (bikallem) [reply CHANGES.md]: removed references to ci

File: cohttp/test/test_header.ml [hunk 3]
```diff
@@ -438,11 +431,6 @@ let link_ext_star () =
       ]
     (H.get_links headers)
 
-let trim_ws () =
```
Comment (mseri) [inline cohttp/test/test_header.ml:441]: Why not keeping this test and running it instead?

File: examples/lwt_unix_doc/docker_lwt.ml [hunk 4]
```diff
@@ -1,14 +1,6 @@
 open Lwt.Infix
 open Cohttp_lwt_unix
 
-let ctx =
```
Comment (mseri) [inline examples/lwt_unix_doc/docker_lwt.ml:4]: We should leave this here and instead replace `t` with
```ocaml
let t =
  Cohttp_lwt_unix.Client.get ~ctx (Uri.of_string "http://docker/version")
  >>= fun (resp, body) ->
  let open Cohttp in
  let code = resp |> Response.status |> Code.code_of_status in
  Printf.printf "Response code: %d\n" code;
  Printf.printf "Headers: %s\n" (resp |> Response.headers |> Header.to_string);
  body |> Cohttp_lwt.Body.to_string >|= fun body ->
  Printf.printf "Body of length: %d\n" (String.length body);
  print_endline ("Received body\n" ^ body)
```
This other body was probably an oversight.
Or at least add a flag to select one of the other behaviour.

These examples were meant to build the examples in the README.
There is a ticket to move to `mdx`, but until that is implemented, we should try and keep them for now.

File: cohttp/test/test_header.ml (hunk 3 above)
Comment (bikallem) [reply cohttp/test/test_header.ml]: Done. Added to the test suite.

File: examples/lwt_unix_doc/docker_lwt.ml (hunk 4 above)
Comment (bikallem) [reply examples/lwt_unix_doc/docker_lwt.ml:4]: Done.

File: cohttp-lwt-unix/test/test_parser.ml [hunk 5]
```diff
@@ -44,15 +44,6 @@ let basic_res_content =
...
    home=Cosby&favorite+flavor=flies"
 
-let post_req =
-  "POST /path/script.cgi HTTP/1.0\r\n\
-   From: frog@jmarshall.com\r\n\
-   User-Agent: HTTPTool/1.0\r\n\
-   Content-Type: application/x-www-form-urlencoded\r\n\
-   Content-Length: 32\r\n\
-   \r\n\
-   home=Cosby&favorite+flavor=flies"
```
Comment (rgrinberg) [inline cohttp-lwt-unix/test/test_parser.ml:54]: What about parsing this request as well?

**Comment by rgrinberg** [comment]
@smorimoto there's some shared cache failures. could you take a look? The shared cache lives on a different FS so hardlinks can't work. We need to either copy the artifacts or use the same FS.

**Comment by rgrinberg** [comment]
Merged.

**Pull request closed without merging**